src/snowflake/snowpark/_internal/data_source/utils.py

```diff
@@ -469,11 +480,18 @@ def process_parquet_queue_with_threads(
 
     if fetch_with_process:
         # Wait for all processes to complete
-        for idx, process in enumerate(workers):
+        for process in workers:
             process.join()
-            if process.exitcode != 0:
+        # empty parquet queue to get all signals after each process ends
+        while not parquet_queue.empty():
```

Comment:
queue.empty is unreliable, https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Queue.empty

if you want to get all elements out of queue, use the 
```python
while True:
    try:
        q.get_nowait or q.get(Flase)
    except queue.Empty:
        break
```